### ISSUE TITLE
<DetailsList> with selectionMode={SelectionMode.single} acts like Radio but is read as checkbox by screen readers

### ISSUE THREAD

**Issue by adityamatt** [microsoft/fluentui#15526]
### Environment Information

- **Package version(s)**: 7.124.2
- **Browser and OS versions**: Chrome and Edge on Windows 10

### Describe the issue:

So If you render a <DetailsList> with selectionMode set as selectionMode.single, it acts like a radio because you can only choose one of the items from the list but it will be read as checkbox by screenreaders, I've tried using onRenderCheckBox property to render a radio, but since the rendered JSX is wrapped inside a div by default that as role "checkbox" it still is read as checkbox.

#### Actual behavior:

 The rendered selection is read as checkbox because the it is wrapped inside a div that has role "checkbox"

#### Expected behavior:

The rendered selection should be read is Radio if selectionMode={SelectionMode.single} is used with DetailsList

### Documentation describing expected behavior

Documentation doesn't refers anywhere on description of radio if selectionMode.single is used
Raising after discussion on Teams channels
[Teams Discussion](https://teams.microsoft.com/l/message/19:151fc127103b4ef389b3dc79d88b08ee@thread.skype/1601875656049?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=6eec88b0-e096-4b28-a631-a6a5ffd97a00&parentMessageId=1601875656049&teamName=CSEO%20Coherence%20Design%20System&channelName=Coherence%20React&createdTime=1601875656049)

**Comment by paulgildea** [comment]
@adityamatt Thanks for the issue.

@smhigley - For the model question

@tomi-msft - work with @smhigley to determine if this is the correct model or if we need to update the model here.
